plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/changedList/ChangedListViewImpl.java

```diff
@@ -172,9 +203,105 @@ public void onClick(ClickEvent event) {
         btnCompare = createButton(locale.buttonCompare(), "git-compare-btn-compare", new ClickHandler() {
             @Override
             public void onClick(ClickEvent event) {
-                delegate.onCompareClicked();
+                delegate.onCompareProvided();
             }
         });
         addButtonToFooter(btnCompare);
     }
-}
\ No newline at end of file
+
+    private List<Node> getGroupedNodes(Map<String, String> items) {
```

Comment:
This method is to complex for understanding, divide it into understandable parts.